### PR TITLE
Override Read/WriteByte on BrotliStream

### PR DESCRIPTION
**Pull request by stephentoub**
The base Stream.Read/WriteByte allocate a byte[1] array and delegate to the corresponding array-based Read/Write methods.  We can do much better on BrolitStream (as we can on most of our Stream-derived types), especially since we already have span-based Read/Write methods, so we can just call those with a span for the single byte.

cc: @ianhays, @joshfree

### CODE REVIEW

File: src/System.IO.Compression.Brotli/tests/BrotliEncoderTests.cs [hunk 1]
```diff
@@ -327,6 +327,46 @@ public void WriteStream(string testFile)
...
+                    Assert.InRange(b, 0, 255);
+                    decompressedBytes[i] = (byte)b;
+                }
+                Assert.Equal(-1, bs.ReadByte());
+                Assert.Equal(-1, bs.ReadByte());
+            }
+
+            for (int i = 0; i < correctUncompressedBytes.Length; i++)
+            {
+                Assert.Equal(correctUncompressedBytes[i], decompressedBytes[i]);
```
Comment (JeremyKuhne) [inline src/System.IO.Compression.Brotli/tests/BrotliEncoderTests.cs]: Is it possible to use Assert.Equal for the arrays directly rather than iterating?

Comment (stephentoub) [reply src/System.IO.Compression.Brotli/tests/BrotliEncoderTests.cs]: Yes, will change.

**Comment by danmosemsft** [comment]
@JeremyKuhne did you mean to sign off?

**Comment by stephentoub** [comment]
@dotnet-bot test OSX x64 Debug Build please (https://github.com/dotnet/corefx/issues/30146)

**Comment by stephentoub** [comment]
@dotnet/dnceng, what's going on with the Linux leg here?  It appears to be hung waiting for a Debian test suite to complete:
https://mc.dot.net/#/user/stephentoub/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/290486404d29792fff39831694c1bcd31de6b431/workItem/System.Net.Http.Unit.Tests
image
but the log shows it's already completed running the tests:
```
2018-06-05 13:37:43,782: INFO: proc(54): run_and_log_output: Output: Discovering: System.Net.Http.Unit.Tests
2018-06-05 13:37:44,101: INFO: proc(54): run_and_log_output: Output: Discovered:  System.Net.Http.Unit.Tests
2018-06-05 13:37:44,290: INFO: proc(54): run_and_log_output: Output: Starting:    System.Net.Http.Unit.Tests
2018-06-05 13:37:53,852: INFO: proc(54): run_and_log_output: Output: Finished:    System.Net.Http.Unit.Tests
2018-06-05 13:37:53,861: INFO: proc(54): run_and_log_output: Output: 
2018-06-05 13:37:53,862: INFO: proc(54): run_and_log_output: Output: === TEST EXECUTION SUMMARY ===
2018-06-05 13:37:53,872: INFO: proc(54): run_and_log_output: Output:    System.Net.Http.Unit.Tests  Total: 1622, Errors: 0, Failed: 0, Skipped: 0, Time: 9.559s
```

**Comment by markwilkie** [comment]
Tracking https://github.com/dotnet/core-eng/issues/3620 to investigate stuck leg

**Comment by sunandabalu** [comment]
Taking a look at the stuck linux leg now.

**Comment by MattGal** [comment]
All the logs and such look totally normal and healthy; I'd suggest focusing your search on the SQL DB if the events for this work item are still there (can save lots of time), or using the eventhub scraping tool if not.

**Comment by JeremyKuhne** [comment]
> @JeremyKuhne did you mean to sign off?

Yes, I trusted @stephentoub to make the requested change and I didn't think it was critical.

**Pull request merged**